models/sessions.go

```diff
@@ -160,17 +162,17 @@ func (s *Session) UpdateAssociatedFactor(tx *storage.Connection, factorID *uuid.
 }
 
 func (s *Session) UpdateAssociatedAAL(tx *storage.Connection, aal string) error {
-	s.AAL = aal
+	s.AAL = &aal
 	return tx.Update(s)
 }
 
 func (s *Session) CalculateAALAndAMR() (aal string, amr []AMREntry) {
 	amr, aal = []AMREntry{}, AAL1.String()
 	for _, claim := range s.AMRClaims {
-		if claim.AuthenticationMethod == TOTPSignIn.String() {
+		if *claim.AuthenticationMethod == TOTPSignIn.String() {
 			aal = AAL2.String()
 		}
-		amr = append(amr, AMREntry{Method: claim.AuthenticationMethod, Timestamp: claim.UpdatedAt.Unix()})
+		amr = append(amr, AMREntry{Method: *claim.AuthenticationMethod, Timestamp: claim.UpdatedAt.Unix()})
```

Comment:
I wonder, if `claim.AuthenticationMethod` is nil will this panic? Can you add a nil check just in case (to skip it if it is nil)?

Reply:
I think I'll just wrap in a helper function as well which will mitigate this issue